voila-gridstack/static/extension.js

```diff
@@ -0,0 +1,243 @@
+            Jupyter.notebook.save_notebook().then(function () {
+
+                // sets notebook to normal width and shows inputs and prompt
+                $('#notebook-container').css('width', '');
+                $('.code_cell > .input').show();
+                $('.prompt').show();
+
+                // removes handles
+                $('.gridhandle').remove();
+
+                // unwraps cells, and removes remaining GridStack elements
+                $('.grid-stack').contents().unwrap();
+                $('.grid-stack-item-content').siblings().remove();
+                $('.grid-stack-item-content').unwrap();
+                $('.grid-stack-item-content').contents().unwrap();
+                $('style[data-gs-style-id]').remove();
+
+                // Remove CSS files
+                $('head').children("link[href='https://cdn.jsdelivr.net/npm/gridstack@0.5.2/dist/gridstack.min.css']").remove();
+                $("head").children("#voila-gridstack-styles").remove();
+
+                // enables button for gristack view
+                $('#btn-voila-gridstack_gridstack').prop( "disabled", false );
+            });
+        }
+
+        /*
+         * Open Gridstack view with handles in Jupyter environment
+         */
+        var open_voila_gridstack = function() {
+
+            // disables button
+            $('#btn-voila-gridstack_gridstack').prop( "disabled", true );
+
+            // hide notebook contents during init
+            $('#site').hide();
+
+            // appends voila-gridstack styles and a loading spinner
+            $("head").append("<link id='voila-gridstack-styles' href='/nbextensions/voila-gridstack/voila-gridstack.css' type='text/css' rel='stylesheet' />").ready( function () {
+                $('#site').after(`
+                    <div id="loading">
+                        <div class="spinner-container">
+                          <svg class="spinner" data-name="c1" version="1.1" viewBox="0 0 500 500" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"><metadata><rdf:RDF><cc:Work rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/><dc:title>voila</dc:title></cc:Work></rdf:RDF></metadata><title>spin</title><path class="voila-spinner-color1" d="m250 405c-85.47 0-155-69.53-155-155s69.53-155 155-155 155 69.53 155 155-69.53 155-155 155zm0-275.5a120.5 120.5 0 1 0 120.5 120.5 120.6 120.6 0 0 0-120.5-120.5z"/><path class="voila-spinner-color2" d="m250 405c-85.47 0-155-69.53-155-155a17.26 17.26 0 1 1 34.51 0 120.6 120.6 0 0 0 120.5 120.5 17.26 17.26 0 1 1 0 34.51z"/></svg>
+                        </div>
+                        <h2 id="loading_text">Running...</h2>
+                    </div>
+                `);
+            });
+
+            // executes all cells
+            Jupyter.notebook.execute_all_cells();
```

Comment:
The grid view hides empty outputs, that can't be known without execution.

I think it should be implement in a next pull request, that may include several additional features : 
- showing grid view without executing all cells
- executing cells from grid view
- choosing which outputs to hide or to show from grid view

Reply:
Hi brichet,

Thanks for the changes so far!

Is there a way to only execute cells that are not yet executed?

Also, I saw that cells that have hidden=true in the metadata, have that property removed when they do have output and the mode is switched to gridstack. Maybe the automatic setting of the hidden property has to many drawbacks?

But you're right, that can be done in a new PR.

Reply:
Hi Mario,

This would be great to only execute cells that are not yet executed, but this means retrieving (i) executed cells on current kernel and (ii) cells which has been modified since last execution.

I'm not sure if this information remains in the notebook.